core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1626,14 +1626,16 @@ public PhysicalOperation visitTopN(TopNNode node, LocalExecutionPlanContext cont
                 sortOrders.add(node.getOrderingScheme().getOrdering(symbol));
             }
 
+            DataSize maxPartialTopNMemorySize = SystemSessionProperties.getMaxPartialTopNMemory(session);
             OperatorFactory operator = TopNOperator.createOperatorFactory(
                     context.getNextOperatorId(),
                     node.getId(),
                     source.getTypes(),
                     (int) node.getCount(),
                     sortChannels,
                     sortOrders,
-                    plannerContext.getTypeOperators());
+                    plannerContext.getTypeOperators(),
+                    node.getStep() == TopNNode.Step.PARTIAL && maxPartialTopNMemorySize.compareTo(DataSize.ofBytes(0)) > 0 ? Optional.of(maxPartialTopNMemorySize) : Optional.empty());
```

Comment:
nit: could you extract this and
```
            Optional<DataSize> maxPartialTopNMemorySize = isPartial ? Optional.of(SystemSessionProperties.getMaxPartialTopNMemory(session)).filter(
                    maxSize -> maxSize.compareTo(DataSize.ofBytes(0)) > 0) : Optional.empty();
```
as separate method?

Reply:
Done